core/src/robot/Robot.cc

```diff
@@ -1442,10 +1442,12 @@ namespace jiminy
 
     hresult_t Robot::refreshTransmissionProxies(void)
     {
+        actuatedJointNames_.clear();
```

Comment:
Just update the list of names of the transmissions here.

Reply:
(because the list of actuated joint names do not need to be refreshed once the transmissions are all initialized.

Reply:
It is better to update in refresh proxies only stuffs that does not depend on the state of the transmissions. So basically not much.

Reply:
why for you the list of actuated joints depends on the state of the transmission ?

Reply:
What I mean by "state" here, I'm talking about the developer point of view not roboticist (more like state machine). If the transmission is not initialized (so the state is post attached to robot but before init) then the transmission is not attached to any joint at this point.

Reply:
By the way, bear in mind that it is allowed to call `initialize` more than one (for instance to recycle to transmission for something else). It is not supposed to happen but not forbidden. So by careful when updating `actuatedJointNames_`. The easy way it is clear it and fill it again as you are already doing.

Reply:
Ok I understand ! thanks; and yes I think it might be the easiest to clear and fully refill